src/does-match.ts

```diff
@@ -0,0 +1,30 @@
+import { deepEquals } from "./deep-equals"
+import { entries } from "./entries"
+import { includes } from "./includes"
+
+export const doesMatch = (target: object, value: object): boolean => {
+
+  const valueKeys = Object.keys(value)
+  const commonKeys = Object.keys(target).filter(value => includes(valueKeys, value))
+  const targetKV = entries(target)
+  const valueKV = entries(value)
+
+  if (targetKV.length === 0)
+    return true
+  else if (commonKeys.length === 0)
+    return false
+  else {
+    const checkTarget: [string, unknown][]= commonKeys.map(commonKey => {
+    return valueKV.find(element => element[0] === commonKey) as [string, unknown] //commonKeys contains all the common keys so each array must contain the key we are looking for
```

Comment:
still alignement errors and useless return

Reply:
Done